### ISSUE TITLE
Wrong selection coordinates in bar chart

### ISSUE THREAD

**Issue by xemlock** [apexcharts/apexcharts.js#2718]
### Description

Apparently for **bar** charts the selection coordinates received by `selection` event handler are off. I was able to calculate proper coordinates by taking into account `w.globals.barPadForNumericAxis` when determining the selection offset (see linked codepen).

I noticed no such problem with **line** charts, the selection coordinates match the visuals there.

### Steps to Reproduce

1. Visit https://codepen.io/xemlock/pen/MWvJKdj
2. Use selection tool to select range
3. Observe incorrect selection coordinates received by `selection` event handler

### Expected Behavior

The selection coordinates received by `selection` event handler match what was selected on the plot 

### Actual Behavior

The selection coordinates received by `selection` event handler don't match what was selected on the plot 

### Screenshots

![Screen Shot 2021-10-23 at 11 17 54](https://user-images.githubusercontent.com/952555/138550684-7babee23-37f1-444b-a78e-6b18b91ea214.png)

### Reproduction Link

https://codepen.io/xemlock/pen/MWvJKdj

**Comment by monk-lee** [comment]
same issue +1

When the mouse leave the charts,
The minimum value is replaced with the value at the time of the first mouse click, and the maximum value is replaced with the value where the mouse leave.

Is there a solution to the problem?

**Comment by jpmcc** [comment]
Similar or same issue with the line chart. However, I notice that if you click within the chart area after making a selection, the min/max values are not replaced when the cursor is moved out.

**Comment by jpmcc** [comment]
Due to the Firefox selection issue #1981 I've had to drop using the selection event. Not sure if it will help anyone, but I am now using the dataPointSelection event - when I get two points, I create a block annotation which gives me the same effect I am after.  If I get more than two points, I discard the first points, clear the block annotation and start again.

**Comment by maxpowel** [comment]
Looks that the bug only appears until the next user click event to the chart. 
If once selection is done you click anywhere inside the chart (including resizing the selection), you can  leave the chart without being affected by this issue.

My findings:
Once the selection is done, a click event is raised so maybe there is a race problem with this event., Maybe it is called too early so the detection feature does not know that the user selection is already done (unless you click again).

My workarround is to store the selection value but only use it when the click event is raised. By using this method, the last "change" when you leave is ignored. I am using angular and this is my code:

```
events: {
        selection: (chartContext, {xaxis, yaxis}) => {
          // Store the value but not use it yet
          chartContext.lastSelection = xaxis
        },
        click: function(event, chartContext, config) {
          if(chartContext.lastSelection != null) {
            // Use the value here
            // mySuperFunction(chartContext.lastSelection)
            console.log(chartContext.lastSelection)
            chartContext.lastSelection = null
          }
        }
      },
```
I will take a look on the code in order to find the problem and send you a pull request with a fix
